lib/util/resolve-base-url.js

```diff
@@ -0,0 +1,16 @@
+'use strict'
+
+var {URL} = require('url')
+
+module.exports = resolveBaseUrl
+
+function resolveBaseUrl(repository) {
+  try {
+    var parsed = new URL(repository)
+
+    return 'https://' + parsed.host
```

Comment:
Could be made into a one liner?

```suggestion
    return 'https://' + new URL(repository).host
```